pkg/fileservice/memcache_stats_collector.go

```diff
@@ -0,0 +1,46 @@
+package fileservice
+
+import (
+	"github.com/matrixorigin/matrixone/pkg/util/metric"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type memcacheStatsCollector struct {
+	memcache *MemCache
+
+	numRead *prometheus.Desc
+	numHit  *prometheus.Desc
+}
+
+// NewMemCacheStatsCollector Creates a Prometheus Batch Stats Collector for MemCache
```

Comment:
Why do we need Prometheus?    Can we just do our own -- I dont think we need this kind of dependency and complexity in our own internal metrics.

Reply:
We heavily use Prometheus in our current code for the fetching different system_metrics: https://github.com/matrixorigin/matrixone/blob/4a1f76759de1edb001aed379f691227db57c6771/pkg/util/metric/m_process.go#L49

Reply:
The current MetricExporter (which collects system_metrics periodically) which we plan to reuse also uses the Prometheus registry. https://github.com/matrixorigin/matrixone/blob/4a1f76759de1edb001aed379f691227db57c6771/pkg/util/metric/mometric/metric.go#L82